maven-extension/README.md

```diff
@@ -0,0 +1,89 @@
+# Maven OpenTelemetry extension
+
+Maven extension to observe Maven builds as distributed traces.
+
+## Getting Started
+
+The Maven OpenTelemetry Extension is configured using environment variables or JVM system properties and can be added to a build using one of the following ways:
+* adding the extension jar to `${maven.home}/lib/ext`
+* adding the path to the extension jar to`-Dmaven.ext.class.path`,
+* adding the extension as a build extension in the `pom.xml`,
+* (since Maven 3.3.1) configuring the extension in `.mvn/extensions.xml`.
+
+
+### Adding the extension to the classpath
+
+Add the Maven OpenTelemetry Extension to `${maven.home}/lib/ext` or to the classpath using `-Dmaven.ext.class.path=`.
+
+```
+cp /path/to/opentelemetry-java-contrib/maven-extension/build/libs/opentelemetry-maven-extension-1.6.0-SNAPSHOT.jar target/dependency/opentelemetry-maven-extension.jar
+export OTEL_EXPORTER_OTLP_ENDPOINT="http://localhost:4317"
+
+mvn -Dmaven.ext.class.path=target/dependency/opentelemetry-maven-extension.jar verify
+```
+
+### Declaring the extension in the `pom.xml` file
+
+WARNING documentation is invalid until the first release of the io.opentelemetry.contrib:opentelemetry-maven-extension is published on a public Maven repository
+
+Add the Maven OpenTelemetry Extension in the `pom.xml` file:
+
+```xml
+<project>
+  ...
+  <build>
+    <extensions>
+      <extension>
+          <groupId>io.opentelemetry.contrib</groupId>
+          <artifactId>opentelemetry-maven-extension</artifactId>
+          <version>1.6.0-SNAPSHOT</version>
```

Comment:
By the way we'll be releasing 1.6.0 this week probably before this PR is merged. I suspect there is no rush to get this into the release but just want to confirm :)

Reply:
I can't say there is a rush but 
* I'll be happy to see this capability released
* We want to archive as soon as possible https://github.com/elastic/opentelemetry-maven-extension